app/lib/ui-state.ts

```diff
@@ -39,10 +40,10 @@ export const getUiState = <
   Data extends Record<string, unknown>,
 >(
   getState: (
-    set: <S extends AvailableStatus, D extends Record<string, unknown>>(
+    set: <S extends AvailableStatus, SData extends Record<string, unknown>>(
```

Comment:
```suggestion
    set: <S extends Status, SData extends Record<string, unknown>>(
```
I think this would allow for more flexibility

Reply:
This update caused issues on status management, we should look at this improvement separately